cmd/ssl-watch/main.go

```diff
@@ -1,87 +1,50 @@
 package main
 
 import (
-	"crypto/tls"
 	"github.com/gorilla/mux"
-	"net"
 	"net/http"
-	"strings"
+	"os"
+	"os/signal"
+	"syscall"
 	"time"
 )
 
-func processDomain(app *App, domain string, ips []net.IP) Endpoints {
+func (app *App) updateMetrics() {
 
-	host, port, err := net.SplitHostPort(domain)
-	if err != nil {
-		host = domain
-		port = "443"
-	}
-
-	if len(ips) == 0 {
-		ips = resolveDomain(app, host, app.config.LookupTimeout)
-	}
-	endpoints := Endpoints{}
-
-	for _, ip := range ips {
-
-		dialer := net.Dialer{Timeout: app.config.ConnectionTimeout, Deadline: time.Now().Add(app.config.ConnectionTimeout + 5*time.Second)}
-
-		if IsIPv4(ip.String()) {
-
-			endpoint := Endpoint{}
-			connection, err := tls.DialWithDialer(&dialer, "tcp", ip.String()+":"+port, &tls.Config{ServerName: host, InsecureSkipVerify: true})
-			if err != nil {
-				app.log.Error(ip.String(), err)
-				endpoint.alive = false
-				endpoints[ip.String()] = endpoint
-				continue
-			}
-
-			cert := connection.ConnectionState().PeerCertificates[0]
-			endpoint.alive = true
-			endpoint.expiry = cert.NotAfter
-			endpoint.CN = cert.Subject.CommonName
-			endpoint.AltNamesCount = len(cert.DNSNames)
-			err = cert.VerifyHostname(host)
-			if err != nil {
-				endpoint.valid = false
-			} else {
-				endpoint.valid = true
-			}
-			connection.Close()
-			endpoints[ip.String()] = endpoint
+	for {
```

Comment:
Why not use something like this:
```	
for ; true; <-ticker.C {
		...
}
```
instead of sleep at the end?

Reply:
that's really nice, thx!